kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryption.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+import io.kroxylicious.proxy.filter.FilterFactory;
+import io.kroxylicious.proxy.plugin.Plugin;
+
+/**
+ * @deprecated Replaced with {@link io.kroxylicious.filter.encryption.RecordEncryption}
+ * A {@link FilterFactory} for {@link RecordEncryptionFilter}.
+ * @param <K> The key reference
+ * @param <E> The type of encrypted DEK
+ */
+@Deprecated(since = "0.5.0", forRemoval = true)
+@Plugin(configType = EnvelopeEncryption.Config.class)
+public class EnvelopeEncryption<K, E> extends RecordEncryption<K, E> implements FilterFactory<EnvelopeEncryption.Config, SharedEncryptionContext<K, E>> {
```

Comment:
can we overload the constructor to log a warning about the deprecated filter?

Reply:
The filter factory already logs the deprecation based on the annotation so we are covered

Reply:
awesome

Reply:
```shell
2024-03-01 13:40:52 WARN  io.kroxylicious.proxy.config.ServiceBasedPluginFactoryRegistry:101 - io.kroxylicious.proxy.filter.FilterFactory plugin with id EnvelopeEncryption is deprecated
```